pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +886,61 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func (r *ReconcileClusterDeployment) selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			if cdNamespace, ok := pod.Labels[install.ClusterDeploymentNamespaceLabel]; ok {
+				retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      cdName,
+					Namespace: cdNamespace,
+				}})
+			}
+		}
+	}
+	return retval
+}
+
+func (r *ReconcileClusterDeployment) updateInstallRetriesTotalMetric(clusterDeploymentName string, clusterDeploymentNamespace string, cdLog log.FieldLogger) {
+	labels := map[string]string{"cluster_deployment": clusterDeploymentName, "namespace": "true"}
+	installerPodLabels := map[string]string{install.ClusterDeploymentNameLabel: clusterDeploymentName, install.ClusterDeploymentNamespaceLabel: clusterDeploymentNamespace, install.InstallJobLabel: "true"}
+	currentRetiesMetric, err := metricClusterDeploymentInstallRetriesTotal.GetMetricWith(labels)
+	if err != nil {
+		cdLog.Error("Error getting cluster deployment install retries total metrics.")
+	} else {
+		cdLog.Infof("CURRENT VALUE:%v", currentRetiesMetric)
+		pods := &corev1.PodList{}
+		err = r.Client.List(context.Background(), client.MatchingLabels(installerPodLabels), pods)
+		if err != nil {
+			log.WithError(err).Error("error listing pods")
+		} else {
+			podRestarts := 0
+			cdLog.Infof("pod length:%v", len(pods.Items))
+			for _, pod := range pods.Items {
+				cdLog.Infof("pod name:%v", pod.Name)
+				for _, cs := range pod.Status.ContainerStatuses {
+					if cs.Name == "installer" {
```

Comment:
It's actually the "hive" pod that will restart, we just run "installer" long enough to copy the binary to the "hive" pod. Could you make this a constant here: https://github.com/openshift/hive/blob/master/pkg/install/generate.go#L229 and re-use the value.

Reply:
weirdly I only see the Restart count fo up in the container named installer and not in hive. 
lastState: {}
    name: hive
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2019-04-04T15:38:37Z
  - containerID: cri-o://38b3fe16544ce3338b03978b724abfd56ab9490d762aea9de3a70d0c4ed97895
    image: registry.svc.ci.openshift.org/openshift/origin-v4.0-2019-04-04-153251@sha256:a2bfd26fa4e88b089267167f3d53c12c58ecb792c337178a3cb33029635a4783
    imageID: registry.svc.ci.openshift.org/openshift/origin-v4.0-2019-04-04-153251@sha256:a2bfd26fa4e88b089267167f3d53c12c58ecb792c337178a3cb33029635a4783
    lastState:
      terminated:
        containerID: cri-o://3e4d87410b4998fc1211d430b9b9aefa0e9e57769b0b305f5281bb925ec3f20b
        exitCode: 1
        finishedAt: 2019-04-04T15:39:10Z
        reason: Error
        startedAt: 2019-04-04T15:39:05Z
    name: installer
    ready: false
    restartCount: 3
    state:
      terminated:

Reply:
This sounds like it may have to do with how you're testing a restart cycle, what did you change to make it fail? 

Changing hiveutil command args might be a better option.